lib/Code/TidyAll/Git/Precommit.pm

```diff
@@ -41,8 +41,8 @@ sub check {
         # Store the stash, and restore it upon exiting this scope
         unless ( $self->no_stash ) {
             run( $self->git_path, "stash", "-q", "--keep-index" );
-            scope_guard { run( $self->git_path, "stash", "pop", "-q" ) };
         }
+        scope_guard { run( $self->git_path, "stash", "pop", "-q" ) unless $self->no_stash };
```

Comment:
Ah, I didn't notice it was using Guard. I think it'd be preferable to switch to Scope::Guard instead. The author of Guard tends to use undocumented Perl internals in scary ways. Scope::Guard is a little less magical but also entirely safe.